droidlet/tools/hitl/nsp_retrain_infra.py

```diff
@@ -0,0 +1,259 @@
+"""
+Copyright (c) Facebook, Inc. and its affiliates.
+"""
+
+import argparse
+import logging
+import os
+import shutil
+import time
+from subprocess import Popen, PIPE, TimeoutExpired
+import boto3
+from datetime import datetime
+
+from typing import List
+
+from utils.data_generator import DataGenerator
+from utils.job_listener import JobListener
+from utils.task_runner import TaskRunner
+
+
+log_formatter = logging.Formatter(
+    "%(asctime)s [%(filename)s:%(lineno)s - %(funcName)s() %(levelname)s]: %(message)s"
+)
+logger = logging.getLogger()
+logger.handlers.clear()
+logger.setLevel("INFO")
+sh = logging.StreamHandler()
+sh.setFormatter(log_formatter)
+logger.addHandler(sh)
+
+LISTENER_SLEEP_TIME = 10  # Seconds to wait before listener looks for new data
+NSP_RETRAIN_TIMEOUT = 18000  # Wait a max of 5h for the NSP retraining script to finish
```

Comment:
do you know how long it will take if trained on our current dataset?

Reply:
On the testing_data.txt file that's currently in S3 it's been taking 30-45 minutes.  I think it takes 2-3x that long with the latest real annotated.txt file.